From 7373c654202149809216cac167c63aa36626fc28 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 24 Jan 2023 16:54:16 +0000
Subject: [PATCH] ARM: dts: Set the LED default-state to "off"

The default-state property for an LED determines the state of the LED
when its trigger is set to "none". Many users expect that specifying a
trigger of "none" means that the LED will be off, and there is a logic
to that assumption, but the current default of "keep" makes the
assumption false.

Given that no Pi models have a default-trigger of "none", changing the
default-state should have no visible impact for most people, but doing
so will permit a more obvious and more consistent way of turning off
the LEDs.

Note that 6.2 included an upstream DT reorganisation that required some
downstream changes, and while doing that I changed the default states.
This commit brings rpi-6.1.y into line with rpi-6.2.y.

See: https://github.com/raspberrypi/firmware/issues/1742

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2708-rpi-b-plus.dts   | 2 ++
 arch/arm/boot/dts/bcm2708-rpi-b-rev1.dts   | 1 +
 arch/arm/boot/dts/bcm2708-rpi-b.dts        | 1 +
 arch/arm/boot/dts/bcm2708-rpi-cm.dtsi      | 1 +
 arch/arm/boot/dts/bcm2708-rpi-zero-w.dts   | 1 +
 arch/arm/boot/dts/bcm2708-rpi-zero.dts     | 1 +
 arch/arm/boot/dts/bcm2709-rpi-2-b.dts      | 2 ++
 arch/arm/boot/dts/bcm2709-rpi-cm2.dts      | 1 +
 arch/arm/boot/dts/bcm2710-rpi-2-b.dts      | 2 ++
 arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts | 2 ++
 arch/arm/boot/dts/bcm2710-rpi-3-b.dts      | 2 ++
 arch/arm/boot/dts/bcm2710-rpi-cm3.dts      | 1 +
 arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dts | 1 +
 arch/arm/boot/dts/bcm2711-rpi-4-b.dts      | 2 ++
 arch/arm/boot/dts/bcm2711-rpi-400.dts      | 3 ++-
 arch/arm/boot/dts/bcm2711-rpi-cm4.dts      | 2 ++
 arch/arm/boot/dts/bcm2711-rpi-cm4s.dts     | 1 +
 17 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
index b3b18a7b5fe9..ebfd7b4092cd 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts
@@ -158,12 +158,14 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 47 0>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "input";
 		gpios = <&gpio 35 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2708-rpi-b-rev1.dts b/arch/arm/boot/dts/bcm2708-rpi-b-rev1.dts
index 50ac75e8d3d9..75cde9a76016 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b-rev1.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-rev1.dts
@@ -176,6 +176,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 16 1>;
 	};
diff --git a/arch/arm/boot/dts/bcm2708-rpi-b.dts b/arch/arm/boot/dts/bcm2708-rpi-b.dts
index 4d7444a31bb6..4a0d98b530b8 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-b.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-b.dts
@@ -158,6 +158,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 16 1>;
 	};
diff --git a/arch/arm/boot/dts/bcm2708-rpi-cm.dtsi b/arch/arm/boot/dts/bcm2708-rpi-cm.dtsi
index c7845d2ba7ff..de4ded34d686 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-cm.dtsi
+++ b/arch/arm/boot/dts/bcm2708-rpi-cm.dtsi
@@ -4,6 +4,7 @@
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 47 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2708-rpi-zero-w.dts b/arch/arm/boot/dts/bcm2708-rpi-zero-w.dts
index 323fa2ebf730..6c239a4ecaf9 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-zero-w.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-zero-w.dts
@@ -214,6 +214,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "actpwr";
 		gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2708-rpi-zero.dts b/arch/arm/boot/dts/bcm2708-rpi-zero.dts
index 406f945d4093..f217bfebb3b7 100644
--- a/arch/arm/boot/dts/bcm2708-rpi-zero.dts
+++ b/arch/arm/boot/dts/bcm2708-rpi-zero.dts
@@ -155,6 +155,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "actpwr";
 		gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
index c6220e6d5fbc..232c206bbbfb 100644
--- a/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
+++ b/arch/arm/boot/dts/bcm2709-rpi-2-b.dts
@@ -158,12 +158,14 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 47 0>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "input";
 		gpios = <&gpio 35 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2709-rpi-cm2.dts b/arch/arm/boot/dts/bcm2709-rpi-cm2.dts
index c9e47c46f4f7..a9e966dc0399 100644
--- a/arch/arm/boot/dts/bcm2709-rpi-cm2.dts
+++ b/arch/arm/boot/dts/bcm2709-rpi-cm2.dts
@@ -192,6 +192,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&virtgpio 0 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2710-rpi-2-b.dts b/arch/arm/boot/dts/bcm2710-rpi-2-b.dts
index c77ff30aa738..11f12c6562cd 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-2-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-2-b.dts
@@ -158,12 +158,14 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 47 0>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "input";
 		gpios = <&gpio 35 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
index 04621bd197c3..e0f0be339992 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
@@ -234,12 +234,14 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 29 0>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "default-on";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
index e0b233562c03..da6069828dde 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -249,12 +249,14 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&virtgpio 0 0>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "input";
 		gpios = <&expgpio 7 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
index 086200a82d3a..c8bc0a8f24c2 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
@@ -191,6 +191,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&virtgpio 0 0>;
 	};
diff --git a/arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dts b/arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dts
index 6522d2aa3d52..8dc7fa4b4b82 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-zero-2-w.dts
@@ -227,6 +227,7 @@ &i2s {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "actpwr";
 		gpios = <&gpio 29 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
index 43a6cdcb7150..28c4a41c1c86 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
@@ -507,12 +507,14 @@ audio_pins: audio_pins {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 42 GPIO_ACTIVE_HIGH>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "default-on";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2711-rpi-400.dts b/arch/arm/boot/dts/bcm2711-rpi-400.dts
index 67df994170c7..9c49e01748a9 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-400.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-400.dts
@@ -514,13 +514,14 @@ audio_pins: audio_pins {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "default-on";
-		default-state = "on";
 		gpios = <&gpio 42 GPIO_ACTIVE_HIGH>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "default-on";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2711-rpi-cm4.dts b/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
index 58a7d541a938..0ef6af181214 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
@@ -519,12 +519,14 @@ audio_pins: audio_pins {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 		gpios = <&gpio 42 GPIO_ACTIVE_HIGH>;
 	};
 
 	pwr_led: led-pwr {
 		label = "led1";
+		default-state = "off";
 		linux,default-trigger = "default-on";
 		gpios = <&expgpio 2 GPIO_ACTIVE_LOW>;
 	};
diff --git a/arch/arm/boot/dts/bcm2711-rpi-cm4s.dts b/arch/arm/boot/dts/bcm2711-rpi-cm4s.dts
index ee6cbabba714..81e6cc67cffe 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4s.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4s.dts
@@ -389,6 +389,7 @@ &ddc1 {
 &leds {
 	act_led: led-act {
 		label = "led0";
+		default-state = "off";
 		linux,default-trigger = "mmc0";
 	};
 };
-- 
2.35.5

